src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.Cipher.cs

```diff
@@ -150,15 +150,9 @@ internal static unsafe bool EvpAeadCipherFinalEx(
             out int bytesWritten,
             out bool authTagMismatch)
         {
-            scoped Span<byte> notNullOutput = output;
-
-            // We can't pass null down to the native shim, so create a valid pointer if we have an empty span.
-            if (notNullOutput.IsEmpty)
-            {
-                notNullOutput = (stackalloc byte[1]).Slice(1);
-            }
-
-            fixed (byte* pOutput = &MemoryMarshal.GetReference(notNullOutput))
+            // We can't pass null down to the native shim, so create a valid pointer if we have an empty span,
```

Comment:
We own the code for the native shim, so this looks like a self-inflicted wound. Would it be better to relax the argument checking in the native shim instead?